ProcessOptimizer/plots.py

```diff
@@ -1,2382 +1,2493 @@
+        for all values.
+        Possible keys:
+        * "interpolation" [string, default=""]: The interplotation visaulization to use for the 2D
+            plots. For possible values, see
+            https://matplotlib.org/stable/gallery/images_contours_and_fields/interpolation_methods.html
+            If empty string, a contour pot is shown. Note that contour plots are not
+            compatible with showing uncertainty.
+        * "uncertain_color"  [any matplotlib color, default=[0,0,0]]: The colour to use
+            for "very" uncertain values in 2D plots.
+        * "colormap" [string, default="viridis_r"]: The colourmap to use. For possible
+            values, see
+            https://matplotlib.org/stable/tutorials/colors/colormaps.html
+        * "normalize_uncertainty" [function with three inputs]: Normalisation function
+            for the uncertainty. First input is a numpy array containing the standard
+            deviations, second and third inputs are the min and max standard deviations'
+            for all 2D plots. Output must be a numpy array of the same dimensions as x
+            with values between 0 and 1. If not, it raises an error, but only if any
+            concrete outputs are outside this range. Output values of 0 shows the color
+            for the expected value modeled objective function. Output values of 1
+            corresponds to the uncertain_color. To not show any uncertainty, use
+            `lambda x, min, max: 0*x`.
+            Another use is to visually deemphasize points with medium uncertainty by e.g
+            `lambda x, global_min, global_max: ((x-global_min)/(global_max-global_min))**(1/2)`.
+            Defaults to
+            `lambda x, global_min, global_max: (x-global_min)/(global_max-global_min)`
+    Raises
+    -------
+    * `ValueError`
+            If pars is not of the expected types.
+
+    Returns
+    -------
+    * `Axes`:
+        The matplotlib axes.
+    """
+    # Setting the options
+    if not oversampling:
+        # Much less oversampling is needed for plotting absolute dependence than for
+        # partial dependence.
+        if usepartialdependence:
+            oversampling = 10
+        else:
+            oversampling = 2
+    if not plot_options:
+        plot_options = {}
+    default_plot_type = {
+        "interpolation": "",
+        "uncertain_color": [0, 0, 0],
+        "colormap": "viridis_r",
+        "normalize_uncertainty": (
+            lambda x, global_min, global_max: (
+                x-global_min)/(global_max-global_min)
+        ),
+    }
+    for k, v in default_plot_type.items():
+        if k not in plot_options.keys():
+            plot_options[k] = v
+    # zscale and levels really belongg in plot_options, but for backwards
+    # compatibility they are given as separate arguments.
+    plot_options["zscale"] = zscale
+    plot_options["levels"] = levels
+
+    x_vals = find_x_vals(pars, result, expected_minimum_samples)
```

Comment:
Maybe add a comment here to the effect of "Identify the location of the point to highlight in the plots".